### PR TITLE
Fix stack overflow in inheritdoc evaluation

### LINKED ISSUE #39158: Using cref to itself inside inheritdoc crashes VS
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/761030/using-cref-to-itself-inside-inheritdoc-crashes-vs.html)._

---
This is a new issue as inheritdoc is just being added in 16.4.0 preview 1. This problem therefor only happens in 16.4.0 preview 1 (or later?). 

With the new feature, we can use /// &lt;inheritdoc/&gt; to define the same xml comments from the inheritted class on members that are inherited. /// &lt;inheritdoc/&gt; works fine, but /// &lt;inheritdoc cref=&quot;MyProperty&quot; /&gt; crashes Visual Studio to desktop ('CTD') upon hovering the mouse over MyProperty. 

To recreate:

1. Create a new .NET Core Console App.
2. Create a new interface: IMyClass with a member (can be property, method, ...) MyMember and add a summary to it with ///.
3. Create a new class: MyClass that implements IMyClass (add &quot;: IMyClass&quot;).
4. Implement the member MyMember. 
5. Add /// &lt;inheritdoc /&gt; above MyMember. So far so good.
6. Change &lt;inheritdoc /&gt; to &lt;inheritdoc cref=&quot;MyMember&quot;/&gt; and move your mouse over MyMember. VS will crash before it can show the tooltip.

---
#### Visual Studio Feedback System on 10/6/2019, 09:24 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 10/8/2019, 01:54 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #39158

### CODE REVIEW

File: src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs [hunk 1]
```diff
@@ -1084,6 +1096,10 @@ private static XNode[] RewriteMany(ISymbol symbol, Compilation compilation, XNod
             {
                 return Array.Empty<XNode>();
             }
+            finally
+            {
+                visitedSymbols.Remove(symbol);
```
Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs]: Why is this needed?

Comment (sharwell) [reply src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs]: Symbols can appear in more than one reference in the same comment. They just can't be nested.

Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs]: This absolutely needs a comment....

Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs]: And to check there's an existing test that includes a symbol twice that triggers this?

Comment (sharwell) [reply src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs]: I'll get the comment and test added for 16.5. preview 1 branch

**Pull request merged**